app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java

```diff
@@ -88,4 +97,134 @@ public void onCurrentLocationClick(Geometry geometry) {
     }
 
     public abstract boolean isSearchMode();
+
+    public ObservableField<String> observeWarningMessage() {
+        observableWarningMessage.set(warning());
+        return observableWarningMessage;
+    }
+
+    public ObservableField<String> observeErrorMessage() {
+        observableErrorMessage.set(error());
+        return observableErrorMessage;
+    }
+
+    public void onLatitudeFocusChanged(boolean hasFocus, String currentLatitudeValue, String errorMessage, Function1<String, Unit> callback) {
+        if (!hasFocus) {
+            if (!currentLatitudeValue.isEmpty()) {
+                Double latString = Double.valueOf(StringExtensionsKt.parseToDouble(currentLatitudeValue));
+                Double lat = DoubleExtensionsKt.truncate(latString);
+                String error;
+                if (!LngLatValidatorKt.isLatitudeValid(lat)) {
+                    error = errorMessage;
+                } else {
+                    error = null;
+                }
+                callback.invoke(lat.toString());
+                observableErrorMessage.set(error);
+            }
+        } else {
+            onItemClick();
+        }
+    }
+
+    public void onLongitudeFocusChanged(boolean hasFocus, String currentLongitudeValue, String errorMessage, Function1<String, Unit> callback) {
+        if (!hasFocus) {
+            if (!currentLongitudeValue.isEmpty()) {
+                Double lonString = Double.valueOf(StringExtensionsKt.parseToDouble(currentLongitudeValue));
+                Double lon = DoubleExtensionsKt.truncate(lonString);
+                String error;
+                if (!LngLatValidatorKt.isLongitudeValid(lon)) {
+                    error = errorMessage;
+                } else {
+                    error = null;
+                }
+                callback.invoke(lon.toString());
+                observableErrorMessage.set(error);
+            }
+        } else {
+            onItemClick();
+        }
+    }
+
+    public boolean isPoint() {
+        return featureType() == FeatureType.POINT;
+    }
+
+    public void onDescriptionClick() {
+        callback.showDialog(label(), description());
+    }
+
+    public Geometry currentGeometry() {
+        if (featureType() == null)
+            throw new NullPointerException("use setFeatureType before setting an initial value");
```

Comment:
I think nobody is catching this throw?

Reply:
True, this was done this way to avoid errors when using the coordinate view outside the form, like in the event initial screen

Reply:
I'll remove it :)